src/Microsoft.AspNetCore.Hosting.Abstractions/IWebHost.cs

```diff
@@ -24,6 +26,13 @@ public interface IWebHost : IDisposable
         /// <summary>
         /// Starts listening on the configured addresses.
         /// </summary>
-        void Start();
+        Task StartAsync(CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Attempt to gracefully stop the host.
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        Task StopAsync(CancellationToken cancellationToken);
```

Comment:
Do you have any tests that calls Stop or StopAsync instead of Dispose and verifies that server is restartable since the hosting and application service providers aren't disposed?